### PR TITLE
Reliability changes

### PR DESCRIPTION
**Pull request by tylerschultz**
## Description
When scanning cluster for Gateways, note any Clusters that were not queryable. Continue on to scan other Clusters for Gateways after encountering an un queryable Cluster.

When converging EndpointSlices to Clusters, skip the deletion of EndpointSlices that correspond to the un-queryable Cluster.

## Related Issues

#68

### CODE REVIEW

File: pkg/controllers/gatewaydns/endpoint_slice_reconciler.go [hunk 1]
```diff
@@ -122,57 +122,67 @@ type ClusterDiff struct {
...
 	}
 
-	for _, item := range existingEndpointSliceMap {
-		if _, ok := desiredEndpointSliceMap[endpointSliceKey(item)]; !ok {
-			clusterDiff.undesired = append(clusterDiff.undesired, item)
+	for _, existingEndpointSlice := range existingEndpointSliceMap {
+		desiredClusterGateway, ok := desiredClusterGatwayMap[EndpointSliceKey(existingEndpointSlice)]
+		if ok {
+			if desiredClusterGateway.Unreachable {
+				log.Info("Skipping delete of undexpected EdpointSlice, unable to query for Gateway's existence", "EndpointSlice", existingEndpointSlice, "Gatewa Cluster", desiredClusterGateway.ClusterName)
```
Comment (christianang) [inline pkg/controllers/gatewaydns/endpoint_slice_reconciler.go:176]: Typos. "undexpected EdpointSlice" and "Gatewa"

File: pkg/controllers/gatewaydns/endpoint_slice_reconciler.go [hunk 2]
```diff
@@ -122,57 +122,67 @@ type ClusterDiff struct {
...
 
-		existingGatewayDNSNamespacedName, ok := item.Annotations[connectivityv1alpha1.GatewayDNSRefAnnotation]
+		existingGatewayDNSNamespacedName, ok := existingEndpointSlice.Annotations[connectivityv1alpha1.GatewayDNSRefAnnotation]
 		if !ok || existingGatewayDNSNamespacedName != gatewayDNSNamespacedName.String() {
 			continue
 		}
 
-		existingEndpointSliceMap[endpointSliceKey(item)] = item
+		existingEndpointSliceMap[EndpointSliceKey(existingEndpointSlice)] = existingEndpointSlice
 	}
```
Comment (rosenhouse) [inline pkg/controllers/gatewaydns/endpoint_slice_reconciler.go:149]: consider extracting to a private method, maybe named `getExistingEndpointSlices`??

**Comment by christianang** [comment]
Tested by spinning up an e2e environment. Stopped the `cluster-a` control plane using `docker stop`. Created a GatewayDNS record on the management cluster. Observed that the `xcc-dns-controller` attempted to talk to `cluster-a`, but failed, continued on the reconcile `cluster-b` and succeeded. Got the list of `EndpointSlices` from `cluster-b` and saw the expected EndpointSlice.

Also deleted the GatewayDNS and observed the EndpointSlice being deleted from `cluster-b`.

**Review by christianang — APPROVED** [review]

**Review by rosenhouse — APPROVED** [review]
LGTM, tiny nits, its all good

**Pull request merged**